test/js-native-api/test_reference_double_free/test_reference_double_free.c

```diff
@@ -44,21 +44,24 @@ static napi_value New(napi_env env, napi_callback_info info) {
 
 static void NoopDeleter(napi_env env, void* data, void* hint) {}
 
-static void DeleteImmediately(napi_env env, napi_callback_info info) {
+static napi_value DeleteImmediately(napi_env env, napi_callback_info info) {
   size_t argc = 1;
   napi_value js_obj;
   napi_ref ref;
+  napi_valuetype type;
+  napi_value undefined;
 
-  NODE_API_CALL_RETURN_VOID(env,
-      napi_get_cb_info(env, info, &argc, &js_obj, NULL, NULL));
+  NODE_API_CALL(env, napi_get_cb_info(env, info, &argc, &js_obj, NULL, NULL));
 
-  napi_valuetype type;
-  NODE_API_CALL_RETURN_VOID(env, napi_typeof(env, js_obj, &type));
+  NODE_API_CALL(env, napi_typeof(env, js_obj, &type));
+  NODE_API_ASSERT(env, type == napi_object, "Expected object parameter");
+
+  NODE_API_CALL(env, napi_wrap(env, js_obj, NULL, NoopDeleter, NULL, &ref));
+  NODE_API_CALL(env, napi_delete_reference(env, ref));
+  NODE_API_CALL(env, napi_remove_wrap(env, js_obj, NULL));
 
-  NODE_API_CALL_RETURN_VOID(env,
-      napi_wrap(env, js_obj, NULL, NoopDeleter, NULL, &ref));
-  NODE_API_CALL_RETURN_VOID(env, napi_delete_reference(env, ref));
-  NODE_API_CALL_RETURN_VOID(env, napi_remove_wrap(env, js_obj, NULL));
+  NODE_API_CALL(env, napi_get_undefined(env, &undefined));
+  return undefined;
```

Comment:
`NULL` can be returned here to indicate no return value, see https://github.com/nodejs/node/blob/main/src/js_native_api_v8.cc#L314.

```suggestion
  return NULL;
```

This pattern is widely used in our tests, like https://github.com/nodejs/node/blob/main/test/js-native-api/3_callbacks/3_callbacks.c#L34.

Reply:
Good point! I changed it to return `NULL`. Thanks!